lib/webserver/WebServer.js

```diff
@@ -233,578 +223,548 @@ const WebServer = function (options) {
+            res.json(data);
+        } catch (err) {
+            res.status(500).send(err.toString());
         }
     });
 
-    this.app.delete("/api/timers/:timerID", function (req, res) {
-        self.vacuum.deleteTimer(req.params.timerID, function (err, data) {
-            if (err) {
-                res.status(500).send(err.toString());
-            } else {
+    this.app.post("/api/set_dnd", async (req, res) => {
+        try {
+            if (req.body && req.body.start_hour && req.body.start_minute && req.body.end_hour && req.body.end_minute) {
+                let data = await this.vacuum.setDndTimer(req.body.start_hour, req.body.start_minute, req.body.end_hour, req.body.end_minute);
                 res.json(data);
+            } else {
+                res.status(400).send("invalid request");
             }
-        });
+        } catch (err) {
+            res.status(500).send(err.toString());
+        }
     });
 
-    this.app.get("/api/get_dnd", function (req, res) {
-        self.vacuum.getDndTimer(function (err, data) {
-            if (err) {
-                res.status(500).send(err.toString());
-            } else {
-                res.json(data);
-            }
-        });
+    this.app.put("/api/delete_dnd", async (req, res) => {
+        try {
+            let data = await this.vacuum.deleteDndTimer();
+            res.json(data);
+        } catch (err) {
+            res.status(500).send(err.toString());
+        }
+
     });
 
-    this.app.post("/api/set_dnd", function (req, res) {
-        if (req.body && req.body.start_hour && req.body.start_minute && req.body.end_hour && req.body.end_minute) {
-            self.vacuum.setDndTimer(req.body.start_hour, req.body.start_minute, req.body.end_hour, req.body.end_minute, function (err, data) {
-                if (err) {
-                    res.status(500).send(err.toString());
-                } else {
-                    res.json(data);
-                }
-            });
-        } else {
-            res.status(400).send("invalid request");
+    this.app.get("/api/get_timezone", async (req, res) => {
+        try {
+            let data = await this.vacuum.getTimezone();
+            res.json(data);
+        } catch (err) {
+            res.status(500).send(err.toString());
         }
+
     });
 
-    this.app.put("/api/delete_dnd", function (req, res) {
-        self.vacuum.deleteDndTimer(function (err, data) {
-            if (err) {
-                res.status(500).send(err.toString());
-            } else {
+    this.app.post("/api/set_timezone", async (req, res) => {
+        try {
+            if (req.body && req.body.new_zone) {
+                let data = await this.vacuum.setTimezone(req.body.new_zone);
                 res.json(data);
+            } else {
+                res.status(400).send("invalid request");
             }
-        });
+        } catch (err) {
+            res.status(500).send(err.toString());
+        }
```

Comment:
This is such a frequent pattern.
Could we use a middleware for this (e.g. as in https://expressjs.com/de/guide/error-handling.html) and avoid all the try-catch blocks alltogether?

Reply:
Good idea, but it seems that it doesn't work with the async functions.
I would leave it that way for now.
If I find a way that works, I send a new PR.